easybuild/toolchains/compiler/gcc.py

```diff
@@ -66,15 +69,19 @@ class Gcc(Compiler):
 
     # used when 'optarch' toolchain option is enabled (and --optarch is not specified)
     COMPILER_OPTIMAL_ARCHITECTURE_OPTION = {
-        systemtools.AMD : 'march=native',
-        systemtools.INTEL : 'march=native',
-        systemtools.POWER: 'mcpu=native',  # no support for march=native on POWER
+        (systemtools.AARCH32, systemtools.ARM): 'mcpu=native', # implies -march=native and -mtune=native
+        (systemtools.AARCH64, systemtools.ARM): 'mcpu=native', # since GCC 6; implies -march=native and -mtune=native
+        (systemtools.POWER, systemtools.POWER): 'mcpu=native',   # no support for -march on POWER; implies -mtune=native
+        (systemtools.X86_64, systemtools.AMD): 'march=native', # implies -mtune=native
+        (systemtools.X86_64, systemtools.INTEL): 'march=native', # implies -mtune=native
```

Comment:
@boegel: These kind of options are a mess in GCC, as they are inconsistent between architectures... `-march=native` is the way to go on x86_64.

On x86_64, `-march=native` implies `-mtune=native`, but `-mcpu` is a deprecated synonym for `-mtune`.  So using `-mcpu=native` would only tune for the host, but not enable all CPU features (as `-march` does).